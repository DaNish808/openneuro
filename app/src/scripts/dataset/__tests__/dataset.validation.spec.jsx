import Validation from '../dataset.validation.jsx';

describe('dataset/Validation', () => {
    const defProps = {
        errors: [],
        warnings: [],
        validating: false,
        display: true
    };
    it('renders successfully', () => {
        const wrapper = shallow(
            <Validation {...defProps} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('renders successfully when validating', () => {
        const wrapper = shallow(
            <Validation {...defProps} validating={true} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('displays nothing with display set to false', () => {
        const wrapper = shallow(
            <Validation {...defProps} display={false} />
        );
        expect(wrapper.html()).toBe(null);
    });
    it('renders when errors is "Invalid" instead of a list', () => {
        const wrapper = shallow(
            <Validation {...defProps} errors={'Invalid'} />
        )
        expect(wrapper).toMatchSnapshot();
    });
});
