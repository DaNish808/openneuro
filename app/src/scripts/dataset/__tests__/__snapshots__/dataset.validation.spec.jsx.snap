// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/Validation does not crash when errors is "Invalid" 1`] = `
<div
  className="fade-in col-xs-12 validation"
>
  <h3
    className="metaheader"
  >
    BIDS Validation
  </h3>
  <Accordion
    activeKey="2"
    className="validation-wrap"
    onSelect={[Function]}
  >
    <Panel
      bsClass="panel"
      bsStyle="default"
      className="status"
      defaultExpanded={false}
      eventKey="1"
      header={
        <div>
          <span
            className="dataset-status ds-danger"
          >
            <i
              className="fa fa-exclamation-circle"
            />
             Invalid
          </span>
          <span
            className="label text-danger pull-right"
          >
             
          </span>
        </div>
      }
    >
      <div>
        This does not appear to be a BIDS dataset
         
      </div>
      <br />
      <ValidationResults
        errors="Invalid"
        warnings={Array []}
      />
    </Panel>
  </Accordion>
</div>
`;

exports[`dataset/Validation renders successfully 1`] = `
<div
  className="fade-in col-xs-12 validation"
>
  <h3
    className="metaheader"
  >
    BIDS Validation
  </h3>
  <Accordion
    activeKey="2"
    className="validation-wrap"
    onSelect={[Function]}
  >
    <Panel
      bsClass="panel"
      bsStyle="default"
      className="status"
      defaultExpanded={false}
      eventKey="1"
      header={
        <div
          className="super-valid"
        >
          <span
            className="dataset-status ds-success"
          >
            <i
              className="fa fa-check-circle"
            />
             Valid
          </span>
        </div>
      }
    >
      <br />
      <ValidationResults
        errors={Array []}
        warnings={Array []}
      />
    </Panel>
  </Accordion>
</div>
`;

exports[`dataset/Validation renders successfully when validating 1`] = `
<div
  className="fade-in col-xs-12 validation"
>
  <h3
    className="metaheader"
  >
    BIDS Validation
  </h3>
  <Spinner
    active={true}
    text="Validating"
  />
</div>
`;
